content/blog/creating-dscc-block-storage-resources-using-dscc-ansible-playbooks.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Creating DSCC block storage resources using DSCC Ansible playbooks
+date: 2023-03-30T14:02:46.043Z
+author: Anusha Y and Sijeesh Kattumunda
+authorimage: /img/Avatar1.svg
+disable: false
+---
+In my previous blog post, I provided an introduction to Ansible playbooks for DSCC and how to use them. In this post, I will show you how to create an Ansible playbook to create block storage resources 
+Use case:
+The objective of this use case is to provision a Primera volume from scratch using DSCC APIs.
+This use case covers the creation of the following resources in the following order:
+
+1. Create a host
+2. Create a host group
+3. Create a volume set
+4. Create volumes
+5. Update the volume set with the volumes created
+6. Export the volume set to the host created
+
+
+**Let’s look at the Ansible playbook resource-wise.**
+
+Variables in Ansible are similar to variables in any programming language that can be declared and used anywhere in the script. In this use case, certain variables will be used in the playbook and these are declared under the ‘vars’ section. One important variable to watch for here is the “config” variable that provides credentials for the APIs to authenticate the request and the host URL to which the request will be sent. The greenlake_config.json file that is mentioned in the value of the config variable will have the clientID, client secret, and the host URL of the DSCC instance.
+
+```yaml
+vars:
+    - config: "{{ playbook_dir }}/greenlake_config.json"
+    - name: Greenlake DSCC volumeset
+    - system_id: 2M29510B8L
+    - initiators: []
+    - host_name: host_from_ansible
+    - host_group_name: hostGroupFromAnsible
+    - operating_system: RHE Linux
+    - workload: ORACLE_LOG
+    - volume_set_name: ansibleVolumeSet
+    - volume_name: ansibleVolume
+```
+
+Contents of greenlake_config.json:
+
+```json
+{
+  "host": "https://us1.data.cloud.hpe.com",
+  "client_id": "009e17ef-c356-4066-8b66-4ee10e1128b1",
+  "client_secret": "53a9dd94d61311ec9b7e3ad2f9067b26"
+}
+```
+
+**Creation of host**
+
+A host represents a physical server and a host group represents a collection of physical servers. In the context of DSCC API, a host is a definition for a group of initiators that belong to a single server and a host group is a group of initiators across servers. 
+
+To create a host, initiator information is needed and this can be obtained by a GET call to get the initiator resources. One of the initiators can be used in the creation of the host.
+This means that the creation of a host requires two calls;, one to get the initiator and another  to create a host. Along with this, you must mention the name of the host, operating system, and a “user_created” flag must be set to true. Variable names will be mentioned between two curly braces and double quotes as shown in the below code snippet.
```

Comment:
Replace ;' with a colon for the sentence: This means that the creation of host requires two calls: one to get the initiator and another to create a host.